File: CVRPTW.Web/ClientApp/CVRPTW/DataSettings/vehicle-routing.model.ts
export class VehicleRoutingModel {
  public bookings: VehicleRoutingModel_Booking[] = [];
  public totalDuration!: number;
  public depot!: VehicleRoutingModel_Depot;
  public center!: VehicleRoutingModel_Location;
  public radius!: number;
}

export class VehicleRoutingModel_Booking {
  public location!: VehicleRoutingModel_Location;
  public serviceFromTime!: Date;
  public serviceToTime!: Date;
  public serviceMins!: number;
  public fuelTypes!: string[];
  public title: string = "";
  public points!: VehicleRoutingModel_Point[];
  public order!: number;
  public nextNodeIndex!: number;
}

export class VehicleRoutingModel_Point {
  public duration!: number;
  public latitude!: number;
  public longitude!: number;
}

export class VehicleRoutingModel_Location {
  public latitude!: number;
  public longitude!: number;
}

export class VehicleRoutingModel_Depot {
  public location: VehicleRoutingModel_Location | undefined;
  public vehicles: VehicleRoutingModel_Depot_Vehicle[] = [];
}

export class VehicleRoutingModel_Depot_Vehicle {
  public name: string = "";
  public fuelTypes: string[] = [];
  public currentLocation: VehicleRoutingModel_Location | undefined;
  public ordinalBookings: VehicleRoutingModel_Booking[] = [];
  public totalDuration: number = 0;
  public containers: VehicleRoutingModel_Deport_Vehicle_Container[] = [];
  public color: string = "#000000";
}

export class VehicleRoutingModel_Deport_Vehicle_Container {
  public fuelTypes: string[] = [];
  public capacity: number | undefined;
}

File: CVRPTW/Models/VehicleRouting/VehicleRoutingModel.cs
using System;
using System.Collections.Generic;

namespace CVRPTW.Models.VehicleRouting
{
    public class VehicleRoutingModel
    {
        public DepotModel Depot { get; set; }
        public BookingModel[] Bookings { get; set; }
        public long TotalDuration { get; set; }

        public Location Center { get; set; }
        public int Radius { get; set; }

        public class BookingModel
        {
            public Location Location { get; set; }
            public DateTime ServiceFromTime { get; set; }
            public DateTime ServiceToTime { get; set; }
            public int ServiceMins { get; set; }
            public FuelType FuelType { get; set; }
            public string Title { get; set; }
            public int[] TimeMatrix { get; set; }
            public List<Point> Points { get; set; }
            public long Order { get; set; }
            public long NextNodeIndex { get; set; }
        }

        public class Point
        {
            public double Latitude { get; set; }
            public double Longitude { get; set; }
            public int Duration { get; set; }
        }

        public class DepotModel
        {
            public Location Location { get; set; }
            public Vehicle[] Vehicles { get; set; }

            public class Vehicle
            {
                public string Name { get; set; }
                public DateTime DeliveryStartTime { get; set; }
                public DateTime DeliveryEndTime { get; set; }
                public Location CurrentLocation { get; set; }
                public BookingModel[] OrdinalBookings { get; set; }
                public long TotalDuration { get; set; }
                public FuelType FuelType { get; set; }
            }

            public int[] TimeMatrix { get; set; }
            public List<Point> Points { get; set; }
            public long NextNodeIndex { get; set; }
        }

        public class Location
        {
            public double Latitude { get; set; }
            public double Longitude { get; set; }
        }
    }
}
